shared/js/initApp.js

```diff
@@ -150,7 +150,43 @@ $(window).on('function_editor_closed', function() {
   }
 })(appOptions.level);
 
+/**
+ * Only execute the given argument if it is a function.
+ * @param callback
+ */
+function callbackSafe(callback, data) {
+  if (typeof callback === 'function') {
+    callback(data);
+  }
+}
+
+dashboard.saveProject = function(callback) { console.log('saving')
+  var app_id = dashboard.currentApp.id;
+  dashboard.currentApp.levelSource = Blockly.Xml.domToText(Blockly.Xml.blockSpaceToDom(Blockly.mainBlockSpace));
+  if (app_id) {
+    storageApps().update(app_id, dashboard.currentApp, function(data) {
+      callbackSafe(callback, data);
+    });
+  } else {
+    storageApps().create(dashboard.currentApp, function(data) {
+      location.hash = data.id;
+      callbackSafe(callback, data);
+    });
+  }
+};
+
 function initApp() {
+  if (appOptions.level.isProject) {
+    if (dashboard.currentApp) {
+      appOptions.level.startBlocks = dashboard.currentApp.levelSource;
+    } else {
+      dashboard.currentApp = {
+        name: 'Untitled'
+      };
+    }
+    $(document).on('mousedown', '#runButton', dashboard.saveProject);
```

Comment:
do we have a rule for when we use mousedown and when we use click (maybe document that in our style guide)? Does this work on mobile?